### PR TITLE
Tag initial values of in parameters of value lists as significant

### PR DESCRIPTION
**Pull request by BerSie**
Introducing a new property InitialValueIsSignificant on ValueListParameterIn and ValueListParameterInOut to expose the information that the initial value, as empty string, is a valid value and should be used e.g. for filtering the value list.

### CODE REVIEW

**Review by RaizoC — DISMISSED** [review]

File: vocabularies/Common.json [hunk 1]
```diff
@@ -867,6 +867,12 @@
             "LocalDataProperty": {
                 "$Type": "Edm.PropertyPath",
                 "@Core.Description": "Path to property that is used to filter the value list with `eq` comparison"
+            },
+            "InitialValueIsSignificant": {
+                "$Type": "Edm.Bool",
+                "$Nullable": true,
```
Comment (ralfhandl) [inline vocabularies/Common.json:873]: @BerSie nullable Boolean? Better Nullable="false" and DefaultValue="false".

**Review by ralfhandl — CHANGES_REQUESTED** [review]

File: vocabularies/Common.json (hunk 1 above)
Comment (BerSie) [reply vocabularies/Common.json]: added Nullable="false" and changed to DefaultValue="false"

File: vocabularies/Common.xml [hunk 2]
```diff
@@ -689,6 +689,10 @@ The minimum structure is
         <Property Name="LocalDataProperty" Type="Edm.PropertyPath" Nullable="false">
           <Annotation Term="Core.Description" String="Path to property that is used to filter the value list with `eq` comparison" />
         </Property>
+        <Property Name="InitialValueIsSignificant" Type="Edm.Bool" Default="false" Nullable="false">
```
Comment (ralfhandl) [inline vocabularies/Common.xml:692]:
```suggestion
        <Property Name="InitialValueIsSignificant" Type="Edm.Boolean" Default="false" Nullable="false">
```

File: vocabularies/Common.xml [hunk 3]
```diff
@@ -700,6 +704,10 @@ The minimum structure is
         <Property Name="LocalDataProperty" Type="Edm.PropertyPath" Nullable="false">
           <Annotation Term="Core.Description" String="Path to property that is used to filter the value list with `startswith` comparison and filled from the picked value list item" />
         </Property>
+        <Property Name="InitialValueIsSignificant" Type="Edm.Bool" Default="false" Nullable="false">
```
Comment (ralfhandl) [inline vocabularies/Common.xml:707]:
```suggestion
        <Property Name="InitialValueIsSignificant" Type="Edm.Boolean" Default="false" Nullable="false">
```

**Review by ralfhandl — CHANGES_REQUESTED** [review]
Unfortunately we don't have a CSDL validator yet ☹️

**Review by RaizoC — APPROVED** [review]

**Review by ralfhandl — APPROVED** [review]

**Review by Vinod-SAP — COMMENTED** [review]
Do we really need InitialValueIsSignificant in ValueListParameterConstant? I mean if someone has annotated ValueListParameterConstant & set the value as empty, it already means this is important isn't it?

**Comment by ralfhandl** [comment]
@Vinod-SAP I only see it in `ValueListParameterIn` and `ValueListParameterInOut`, and we need it in both.

**Comment by Vinod-SAP** [comment]
@ralfhandl, You are correct. My bad :(

**Comment by ralfhandl** [comment]
@Vinod-SAP No worries, the Github preview of the changes was really misleading 😄

**Pull request merged**